base_model_restrict_update/models/ir_model_access.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2021 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models, tools
+from odoo.exceptions import AccessError
+from odoo.tools.translate import _
+
+
+class IrModelAccess(models.Model):
+    _inherit = "ir.model.access"
+
+    @api.model
+    @tools.ormcache_context(
+        "self._uid", "model", "mode", "raise_exception", keys=("lang",)
+    )
+    def check(self, model, mode="read", raise_exception=True):
+        res = super(IrModelAccess, self).check(model, mode, raise_exception)
+        if self._uid == 1:
+            return True
```

Comment:
```suggestion
        if self.env.su:
            return True
        res = super().check(model, mode, raise_exception)
```
This may be slightly better.